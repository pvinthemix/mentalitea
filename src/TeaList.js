import React from 'react';
import './styles/Main.scss';
import TeaCard from './TeaCard.js';


export default function TeaList(props) {
    return (
     <div className="tea-list">
        <h1>Teas for when you're feeling <span className="selected-descriptor">{props.userSelectedMood}</span>:</h1>
        <ul>
        {
            props.userSelectedTea.map((tea, index) => {
                return (
                    <TeaCard key={index} filteredTeaList={tea}/>
                )
            }) 
        }
        </ul>
     </div>
    );
}