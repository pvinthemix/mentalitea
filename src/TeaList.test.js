import React from 'react';
import TeaList from './TeaList';
import { shallow } from 'enzyme';

const userSelectedTeas = [{
  tea: "Irish Breakfast",
  category: "Black",
  moodId: 5,
  caffeine: "high",
  tastes: [
  "citrus",
  "spice",
  "malty"
  ],
  link: "https://www.adagio.com/black/irish_breakfast.html"
  },
  {
  tea: "Earl Grey Bravo",
  category: "Black",
  moodId: 5,
  caffeine: "high",
  tastes: [
  "citrus",
  "zesty",
  "sweet"
  ],
  link: "https://www.adagio.com/black/earl_grey_bravo.html"
  },
  {
  tea: "Black Dragon Pearls",
  category: "Black",
  moodId: 5,
  caffeine: "high",
  tastes: [
  "chocolate",
  "sweet",
  "earthy"
  ],
  link: "https://www.adagio.com/black/black_dragon_pearls.html"
  }];

  const userSelectedMood = [];

  describe('teaList', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(
        <TeaList userSelectedTea={userSelectedTeas}
                 userSelectedMood={userSelectedMood}
        />);
    });

    it('should display all tea cards', () => {
      expect(wrapper).toMatchSnapshot();
    });

  });