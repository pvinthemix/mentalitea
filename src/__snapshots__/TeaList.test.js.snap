// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`teaList should display all tea cards 1`] = `
<div
  className="tea-list"
>
  <div
    className="tealist-heading-container"
  >
    <h1
      className="tealist-heading"
    >
      Teas for when you're feeling 
      <span
        className="selected-descriptor"
      />
       :
    </h1>
  </div>
  <ul>
    <TeaCard
      filteredTeaList={
        Object {
          "caffeine": "high",
          "category": "Black",
          "link": "https://www.adagio.com/black/irish_breakfast.html",
          "moodId": 5,
          "tastes": Array [
            "citrus",
            "spice",
            "malty",
          ],
          "tea": "Irish Breakfast",
        }
      }
      id="Irish Breakfast"
      key="0"
    />
    <TeaCard
      filteredTeaList={
        Object {
          "caffeine": "high",
          "category": "Black",
          "link": "https://www.adagio.com/black/earl_grey_bravo.html",
          "moodId": 5,
          "tastes": Array [
            "citrus",
            "zesty",
            "sweet",
          ],
          "tea": "Earl Grey Bravo",
        }
      }
      id="Earl Grey Bravo"
      key="1"
    />
    <TeaCard
      filteredTeaList={
        Object {
          "caffeine": "high",
          "category": "Black",
          "link": "https://www.adagio.com/black/black_dragon_pearls.html",
          "moodId": 5,
          "tastes": Array [
            "chocolate",
            "sweet",
            "earthy",
          ],
          "tea": "Black Dragon Pearls",
        }
      }
      id="Black Dragon Pearls"
      key="2"
    />
  </ul>
</div>
`;
